department-of-veterans-affairs/caseflow
# CO Non-Availability Sheet Validations

Connects #5773 

### Description
This PR adds the scaffolding to the hearing spreadsheet validations. It also includes the logic for the CO non-availability spreadsheet validations.

### Testing Plan
1. Download the RO Assignments template, fill in the CO sheet with dates, and save it
1. In the rails console, run S3Service::store_file(<file_name>, <file_path>, :filepath)
1. Create a schedule period with the following command: 'RoSchedulePeriod.new(user: User.first, start_date: 90.days.ago, end_date: 1.days.ago, file_name: <file_name>).save!
1. Run 'SchedulePeriod.first.validate_spreadsheet', and confirm you get the expected result
1. Repeat with various errors in the spreadsheet
